.github/workflows/ci.yml

```diff
@@ -15,19 +15,21 @@ jobs:
         os: [ubuntu-latest, macos-latest, windows-latest]
     runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           submodules: recursive
-      - uses: actions/setup-python@v2
+      - uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
+          cache: 'poetry'
       - name: Run image
         uses: abatilo/actions-poetry@v2.3.0
       - name: Install
-        run: poetry install
+        run: poetry install && echo "$(poetry env info --path)/bin" >> $GITHUB_PATH
       - name: Test
         run: poetry run pytest -v
       - name: Fmt
         run: poetry run black . --check
       - name: Flake8
         run: poetry run flake8
+      - uses: jakebailey/pyright-action@v1
```

Comment:
Can you instead add it to the pyproject.toml and use it through poetry like the other tools?

Reply:
unfortunately pyright is in typescript, so it can't easily managed through poetry. I think setting up `pre-commit` here would be ideal, but that it a little outside the scope of this since that entails a few bigger changes

Reply:
🤧 forgot that it wasn't written in Python